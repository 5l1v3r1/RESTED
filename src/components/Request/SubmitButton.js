import React, { PropTypes } from 'react';
import { Button } from 'react-bootstrap';

function SubmitButton(props) {
  if (props.editMode) {
    return (
      <Button type="submit" bsStyle="success">
        Update request
      </Button>
    );
  }

  return (
    <Button type="submit" bsStyle="primary">
      Send request
    </Button>
  );
}

SubmitButton.propTypes = {
  editMode: PropTypes.bool.isRequired,
  /* eslint-enable react/forbid-prop-types */
};

export default SubmitButton;

