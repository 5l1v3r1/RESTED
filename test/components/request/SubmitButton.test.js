import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

/* eslint-disable import/no-unresolved */
import SubmitButton from 'components/Request/SubmitButton';

describe('SubmitButton', () => {
  describe('EditMode: false', () => {
    let initialProps;

    beforeEach(() => {
      initialProps = {
        editMode: false,
      };
    });

    it('should match the previous snapshot with editMode false', () => {
      const tree = renderer
        .create(<SubmitButton {...initialProps} />);

      expect(tree).toMatchSnapshot();
    });

    it('should render a submit button with a New Request Label', () => {
      const tree = mount(
        <SubmitButton {...initialProps} />,
      );

      expect(tree.find('button').length).toBe(1);
      expect(tree.find('button').text()).toBe('Send request');
    });
  });

  describe('EditMode: true', () => {
    let initialProps;

    beforeEach(() => {
      initialProps = {
        editMode: true,
      };
    });

    it('should match the previous snapshot with editMode true', () => {
      const tree = renderer
        .create(<SubmitButton {...initialProps} />);

      expect(tree).toMatchSnapshot();
    });

    it('should render a submit button with Update label', () => {
      const tree = mount(
        <SubmitButton {...initialProps} />,
      );
      expect(tree.find('button').length).toBe(1);
      expect(tree.find('button').text()).toBe('Update request');
    });
  });

  describe('Compact: true', () => {
    let initialProps;

    beforeEach(() => {
      initialProps = {
        editMode: false,
        compact: true,
      };
    });

    it('should match the previous snapshot with compact true', () => {
      const tree = renderer
        .create(<SubmitButton {...initialProps} />);

      expect(tree).toMatchSnapshot();
    });

    it('should render a submit button with a Compact Request Label', () => {
      const tree = mount(
        <SubmitButton {...initialProps} />,
      );

      expect(tree.find('button').length).toBe(1);
      expect(tree.find('button').text()).toBe('Send');
    });
  });
});

