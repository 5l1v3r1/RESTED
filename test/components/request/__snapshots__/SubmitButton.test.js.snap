// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton EditMode: false should match the previous snapshot with editMode false 1`] = `
<button
  className="btn btn-primary"
  disabled={false}
  type="submit"
>
  Send request
</button>
`;

exports[`SubmitButton EditMode: true should match the previous snapshot with editMode true 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  type="submit"
>
  Update request
</button>
`;
